app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java

```diff
@@ -313,6 +347,114 @@ public static void setAnomaly_ndt(String entry, NetworkMeasurement test){
         }
     }
 
+    /*
+     whatsapp: red if "whatsapp_endpoints_status" or "whatsapp_web_status" or "registration_server" are "blocked"
+     docs: https://github.com/TheTorProject/ooni-spec/blob/master/test-specs/ts-018-whatsapp.md#semantics
+     */
+    public static void setAnomaly_whatsapp(String entry, NetworkMeasurement test) {
```

Comment:
My eyes! `mixedCamel_and_snake_Case`

Reply:
ehehe because is was using functionName when it was only setAnomaly, then i added test_name 😅

Reply:
As every function is mixedCamel, what do you suggest to use for them? to use testName even if everywhere the vars are called test_name? @hellais

Reply:
I _think_ we might want to use `camelCase` for methods, `PascalCase` for classes. IIRC on Java usually people uses `camelCase` also for attributes. I would say we can do more or less what we want with private attributes, as long as we strive for consistency.

Reply:
I even believe the setAnomaly function should be only one with test name as param

Reply:
I think the correct name for a function like this one should be `setAnomalyWhatsapp`

Reply:
and for `setAnomaly_facebook_messenger` ? should I use `setAnomalyFacebookMessenger` even if the test is called `facebook_messenger`?

Reply:
Yes correct `setAnomalyFacebookMessenger`, well the test is technically called "Facebook Messenger", `facebook_messenger` is the snake case version of it, while `facebookMessenger` is the camelCase capitalisation of it.